src/Constraint/Constraint.php

```diff
@@ -128,13 +151,18 @@ public function versionCompare($a, $b, $operator, $compareBranches = false)
     }
 
     /**
-     * @param Constraint $provider
+     * @param ConstraintInterface $provider
      * @param bool $compareBranches
      *
      * @return bool
      */
-    public function matchSpecific(Constraint $provider, $compareBranches = false)
+    public function matches(ConstraintInterface $provider, $compareBranches = false)
     {
+        if ($provider instanceof MultiConstraint) {
+            // turn matching around to find a match
+            return $provider->matches($this);
+        }
```

Comment:
Stating the obvious doesn't really help. But if you can tell me what the expected behaviour used to be I will gladly write some tests and make sure that it continues to behave properly.

I'm guessing it should always return true. Because $emptyConstraint->matches($anything) returns true and $constraint->matches($emptyConstraint) ends up in the implicit return true branch.